File: go/tests/test_prefix128.go

import "math/big"

import "./ipaddress"

// use std::collections::HashMap;


    type Prefix128Test struct {
        pub u128_hash: HashMap<usize, BigUint>,
    }

    func setup() Prefix128Test {
        p128t := Prefix128Test { u128_hash: HashMap::new() };
        p128t.u128_hash.insert(32,
                               BigUint::parse_bytes(b"340282366841710300949110269838224261120",
                                                    10)
                                   .unwrap());
        p128t.u128_hash.insert(64,
                               BigUint::parse_bytes(b"340282366920938463444927863358058659840",
                                                    10)
                                   .unwrap());
        p128t.u128_hash.insert(96,
                               BigUint::parse_bytes(b"340282366920938463463374607427473244160",
                                                    10)
                                   .unwrap());
        p128t.u128_hash.insert(126,
                               BigUint::parse_bytes(b"340282366920938463463374607431768211452",
                                                    10)
                                   .unwrap());
        return p128t;
    }

int main() {
  describe("", func() {
    it ("test_initialize", func() {
        assert!(prefix128::new(129).is_err());
        assert!(prefix128::new(64).is_ok());
    })

    it ("test_method_bits", func() {
        let prefix = prefix128::new(64).unwrap();
        let mut str = String::new();
        for i in 0..64 {
            str.push_str("1");
        }
        for i in 0..64 {
            str.push_str("0");
        }
        assert_eq!(str, prefix.bits())
    })

    it ("test_method_to_u32", func() {
        for (num, u128) in setup().u128_hash {
            assert_eq!(u128, prefix128::new(num).unwrap().netmask())
        }
    })
  })
}

File: python/tests/test_prefix128.py
from ip_address.prefix128 import Prefix128


class Prefix128Test:
    def __init__(self):
        self.u128_hash = []


def setup():
    p128t = Prefix128Test()
    p128t.u128_hash.append((32, int("340282366841710300949110269838224261120")))
    p128t.u128_hash.append((64, int("340282366920938463444927863358058659840")))
    p128t.u128_hash.append((96, int("340282366920938463463374607427473244160")))
    p128t.u128_hash.append((126, int("340282366920938463463374607431768211452")))
    return p128t


def test_initialize():
    assert not Prefix128.create(129)
    assert Prefix128.create(64)


def test_method_bits():
    prefix = Prefix128.create(64)
    expected = "1" * 64 + "0" * 64
    assert expected == prefix.bits()


def test_method_to_u32():
    for num, expected in setup().u128_hash:
        assert expected == Prefix128.create(num).netmask()
